scripts/build_salt_cavern_potentials.py

```diff
@@ -0,0 +1,504 @@
+
+        if "brine" in dep:
+            return "excluded_brine"
+        if "halokinetic" in dep or "halokinetic" in geo:
+            return "dome"
+        if (
+            "stratabound" in dep
+            or "evaporite" in dep
+            or "stratabound" in geo
+            or "evaporite" in geo
+        ):
+            return "bedded"
+        return "unknown"
+
+    gdf["salt_type"] = gdf.apply(classify, axis=1)
+    return gdf["salt_type"]
+
+
+def apply_landuse_exclusions(gdf, regions):
+    """
+    Applies land use exclusion zones to a GeoDataFrame of underground salt storage areas
+    by removing regions intersecting with specified Copernicus land use types.
+
+    The function performs the following steps:
+    1. Applies a negative buffer to each salt storage geometry, with buffer size depending
+       on whether the salt type is 'bedded' or 'dome'.
+    2. Loads Copernicus land cover data and extracts raster cells matching specified grid codes.
+    3. Converts matching raster regions into vector geometries and applies optional buffers.
+    4. Merges all exclusion geometries and removes intersecting salt storage areas.
+
+    Parameters
+    ----------
+    gdf : geopandas.GeoDataFrame
+        GeoDataFrame containing underground salt storage areas with a column 'salt_type'
+        and geometries in any CRS.
+    area_crs : str or pyproj.CRS
+        Target coordinate reference system (CRS) in which geometries will be processed and returned.
+
+    Returns
+    -------
+    geopandas.GeoDataFrame
+        A filtered GeoDataFrame in the specified CRS, excluding areas that intersect with
+        Copernicus land use exclusions.
+
+    Notes
+    -----
+    - Buffers for 'bedded' and 'dome' salt types, as well as Copernicus grid codes and
+      optional per-code buffer distances, are loaded from `snakemake.params.underground_storage`.
+    - This function assumes that `paths.copernicus` and `distance_crs` are defined elsewhere
+      in the Snakemake workflow context.
+    """
+    cop = snakemake.params.underground_storage["copernicus"]
+    salt_buffer_m_bedded = snakemake.params.underground_storage["salt_buffer_m_bedded"]
+    salt_buffer_m_dome = snakemake.params.underground_storage["salt_buffer_m_dome"]
+
+    gdf = gdf.to_crs(distance_crs).copy()
+
+    # Apply buffer by salt type before masking
+    gdf["geometry"] = gdf.apply(
+        lambda row: row.geometry.buffer(
+            -salt_buffer_m_bedded if row.salt_type == "bedded" else -salt_buffer_m_dome
+        ),
+        axis=1,
+    )
+    gdf = gdf[gdf.is_valid & ~gdf.is_empty]
+
+    exclusion_shapes = []
+
+    # Loading Copernicus raster
+    with rioxarray.open_rasterio(paths.copernicus, masked=True) as rds:
+        rds_clip = rds.squeeze().rio.clip_box(*regions.total_bounds)
+        rds_reproj = rds_clip.rio.reproject(distance_crs)
+        clc = rds_reproj.load()
+        transform = clc.rio.transform()
+
+        for code in cop["grid_codes"]:
```

Comment:
What are the performances for the cases you run?
This loop combined with multiple codes and the unary_union may require some for large regions and the unary_union may lead to some computational time, though this issue may be faced when the problem arises.